### ISSUE TITLE
Plugin not initializing on Android

### ISSUE THREAD

**Issue by timricker** [NativeScript/push-plugin#44]
The following line inside `push-plugin.android.js`:

`com.telerik.pushplugin.PushLifecycleCallbacks.registerCallbacks(app.android.nativeApp);`

is causing the plugin to throw an error by the underlying telerik component:

`The application is null, it's not passed correctly!`

It appears that `app.android.nativeApp` is undefined/null at the time of the plugin being initialized.

Is anyone else experiencing this issue?

@yyosifov, would you be so kind to take a look? Thanks!

**Comment by svzi** [comment]
I experience the same issue as well. I'm using TNS 2.3 and TypeScript (but no Angular2) for my project.

**Comment by timricker** [comment]
hey @svzi, the issue is that the plugin is loading before the `nativeApp` has been initialized. You can resolve the issue by importing the push plugin *after* the app has initialized. 

I simply moved this line:
`var pushPlugin = require('nativescript-push-notifications')` to inside the method that needed to use it which in my case is called after init so no longer has the issue.

Also take a look at this for another approach using `application.launchEvent`:
https://github.com/NativeScript/push-plugin/issues/25

**Comment by svzi** [comment]
hey @timricker! Thank you very much for your support, you really saved my day. 👍 

Simply moving the line didn't resolve it in my scenario, but the approach using `application.launchEvent` did it.

**Comment by svzi** [comment]
For whatever reason it didn't work for me. I've stumbled upon other issues hereafter, so removed the plugin from my project and re-added it.

It's the same line of code that you mentioned 7 days ago, but I found out that for me it's not the `app.android.nativeApp` is null, but instead `com.telerik.pushplugin` is undefined. 

The only line of code I added to my project is `var pushPlugin = require("nativescript-push-notifications");` and I call it during the `application.launchEvent`. Why is `com.telerik.pushplugin` undefined at that stage? Why it's undefined at all?

Any ideas?

**Comment by AntonDobrev** [comment]
@svzi Can you please check how are you adding the dependency (e.g. the NPM package of the plugin), the problem you are encountering most probably signiifes that the dependency is not added correctly.

**Comment by yairvillarp** [comment]
Im having the same problem, as @svzi, the `com.telerik.pushplugin` is undefined and for me the error is:

> cannot read property 'PushLifecycleCallbacks' of undefined

on this line

com.telerik.pushplugin.PushLifecycleCallbacks.registerCallbacks(app.android.nativeApp);

**Comment by svzi** [comment]
@AntonDobrev Sorry, I was on a vacation last week. So my response is somewhat delayed. 

I'm not really sure how I can check what you want me to check. Could you please advise? What do you need exactly? The dependency version from my package.json?

Btw I found a workaround for my issue. I just need to call the require-statement async with setTimeout like this:

`setTimeout(function () {
      var pushPlugin = require("nativescript-push-notifications");
      ...     
`

**Comment by AntonDobrev** [comment]
@svzi @yairvillarp This is indeed peculiar, but the only case where I was able to reproduce it was when the dependency was added manually and contained only the NS module w/o the native implementation. That is why my suggestion would be to:
- Remove the push plugin
- Remove android platform
- Add Android platform 
- Add the push plugin from npm 

If the problem persists, let me know of the exact steps to reproduce and the working environment or provide a sample project that illustrates it. Thanks in advance.
